### PR TITLE
Merge labels from KafkaCluster resource to child objects

### PR DESCRIPTION
**Pull request by sebdoido**
| Q               | A
| --------------- | ---
| Bug fix?        | no|yes
| New feature?    | no|yes
| API breaks?     | no|yes
| Deprecations?   | no|yes
| Related tickets | fixes #X, partially #Y, mentioned in #Z
| License         | Apache 2.0

### What's in this PR?

Update util.MergeLabels to a variadic Function to merge multiple map of labels.

### Why?

It could be great that "child" objects (eg. pods, services, cm, pvc) inherit labels from KafkaCluster resource.

### Additional context

This docker image is available if you want to test the feature
`sebdoido/kafka-operator:0.8.1-4`

```
helm upgrade -i kafka-operator --namespace=dbre banzaicloud-stable/kafka-operator --set operator.image.repository=sebdoido/kafka-operator --set operator.image.tag=0.8.1-4
```
### Checklist

- [ ] Implementation tested
- [ ] Error handling code meets the [guideline](https://github.com/banzaicloud/pipeline/blob/master/docs/error-handling-guide.md)
- [ ] Logging code meets the guideline
- [ ] User guide and development docs updated (if needed)

### To Do

- [ ] If the PR is not complete but you want to discuss the approach, list what remains to be done here

### CODE REVIEW

**Comment by baluchicken** [comment]
We really like the idea that the child objects can inherit labels from the KafkaCluster resource. Although we think it should be optional. We recommend to add a new field to the `kafkacluster_types` something like `ClusterLabels` and if that one is set the operator should place the labels in all resources. What do you think?

**Comment by sebdoido** [comment]
Thanks for your comment.
You're right, i'm going to continue to work on this PR on that way, adding a flag to feature-flip this behaviour.

**Comment by sebdoido** [comment]
Hi there, 
image `sebdoido/kafka-operator:0.8.1-5` is available if you want to test the latest commit.

Need to add this flag to your manifest `propagateLabels: true` 
Example:
```
config/samples/simplekafkacluster.yaml
11:  propagateLabels: true
```

**Comment by tinyzimmer** [comment]
From a quick look, it appears you need to do a `make generate manifests` and commit the result.

The `make manifests` will update the CRD in `config/base/crds`, and it will have to be manually added to the copy in the helm chart.

**Comment by sebdoido** [comment]
Thanks, i updated the pr

File: api/v1beta1/kafkacluster_types.go [hunk 1]
```diff
@@ -36,6 +36,7 @@ type KafkaClusterSpec struct {
 	Brokers                []Broker                `json:"brokers"`
 	RollingUpgradeConfig   RollingUpgradeConfig    `json:"rollingUpgradeConfig"`
 	OneBrokerPerNode       bool                    `json:"oneBrokerPerNode"`
+	PropagateLabels        bool                    `json:"propagateLabels"`
```
Comment (tinyzimmer) [inline api/v1beta1/kafkacluster_types.go]: I think we should add an `omitempty` tag here. So it defaults to `false` without bothering the user.  Then the next `make manifests` will remove it from the required fields.

**Comment by sebdoido** [comment]
Thanks, good catch

**Comment by sebdoido** [comment]
`I see a few other places where we should probably do that too` 
> Let me know if you want to I complete some stuff through this PR ?

**Comment by tinyzimmer** [comment]
Don't worry about that, for this I'd say just make sure the helm chart is updated.

**Comment by sebdoido** [comment]
Hi there
any chance to get this PR merged? 
Thanks

**Comment by baluchicken** [comment]
Sure will check it today

**Comment by baluchicken** [comment]
Thanks for the contribution @sebdoido

**Pull request merged**